lib/metadata/acl.js

```diff
@@ -41,8 +42,21 @@ const acl = {
             /* eslint-disable no-param-reassign */
             objectMD.acl = addACLParams;
             objectMD.originOp = 's3:ObjectAcl:Put';
-            return metadata.putObjectMD(bucket.getName(), objectKey, objectMD, params, log,
-                cb);
+
+            // Use storageType to determine if replication update is needed, as it is set only for
+            // "cloud" locations. This  ensures that we reset replication when CRR is used, but not
+            // when multi-backend replication (i.e. Zenko) is used.
+            // TODO: this should be refactored to properly update the replication info, accounting
+            // for multiple rules and resetting the status only if needed CLDSRV-646
+            const replicationInfo = getReplicationInfo(objectKey, bucket, true);
```

Comment:
Note for later: CLDSRV-647 is slated to add an extra parameter to this function call (to provide the S3 config explicitly), but it hasn't been reviewed yet so it may change.